cors/304.htm

```diff
@@ -0,0 +1,105 @@
+<!DOCTYPE html>
+<meta charset=utf-8>
+<title>CORS - 304 Responses</title>
+<meta name=author title="Mark Nottingham" href="mailto:mnot@mnot.net">
+
+<script src=/resources/testharness.js></script>
+<script src=/resources/testharnessreport.js></script>
+<script src=support.js?pipe=sub></script>
+
+<h1>CORS - 304 Responses</h1>
+<div id=log></div>
+<script>
+
+
+/*
+ * 304 Responses
+ */
+
+// A header used to correlate requests and responses
+var state_header = "content-language"
+
+function req(url, id, t, ready) {
+  console.log('starting request ' + id)
+  var client = new XMLHttpRequest()
+  client.error_caught = false
+  try {
+    client.open('GET', url, true)
+    client.setRequestHeader(state_header, id)
+    client.send()
+  } catch (e) {
+    client.error_caught = e
+  }
+  client.onreadystatechange = function() {
+    if (client.readyState == client.DONE) {
+      console.log('finished request ' + id)
+      t.step(function() {
+        assert_true(client.status != 299, "req " + id + " server says: " + client.responseText)
```

Comment:
This is just testing for a server-specific error, denoted by 299 (yes, I realise how hypocritical that makes me...). The actual status code expected depends upon the specific test being run.